homeassistant/components/idasen_desk/button.py

```diff
@@ -0,0 +1,74 @@
+"""Representation of Idasen Desk buttons."""
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_NAME, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DeskData, IdasenDeskCoordinator
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Idasen Desk connection buttons from config entry."""
+    data: DeskData = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            IdasenDeskConnectButton(data.address, data.device_info, data.coordinator),
+            IdasenDeskDisconnectButton(
+                data.address, data.device_info, data.coordinator
+            ),
+        ]
+    )
+
+
+class IdasenDeskConnectButton(ButtonEntity):
```

Comment:
If you plan on adding more buttons in the future it might be easier to use ButtonEntityDescription and add a press_fn


There are a few examples in the codebase already of this. Let me know if you need help finding them

Reply:
Done, I think. Let me know if that is not what you had in mind